src/main/java/gov/nasa/cumulus/CnmToGranuleHandler.java

```diff
@@ -133,6 +135,25 @@ public String PerformFunction(String input, Context context) throws Exception {
 		granule.addProperty("granuleId", granuleId);
 
 		JsonArray inputFiles = cnmObject.getAsJsonObject("product").getAsJsonArray("files");
+		
+		// 'files' Json array does not exist. Initialize
+		if (inputFiles == null) {
+			inputFiles = new JsonArray();
+		}
+		
+		// Add files from each list of files in "filegroups"
+		StreamSupport.stream(cnmObject.getAsJsonObject("product")
+				.getAsJsonArray("filegroups")
```

Comment:
Getting NPE if filegroups doesn't exist in the CNM.

Reply:
My mistake, I thought the original test was passing but I must have changed something since running that test. I'll fix this issue

Reply:
Fixed, the handler will now only try to get `filegroups` from the CNM if `files` is not present